tests/integration/gateway/main_test.go

```diff
@@ -0,0 +1,148 @@
+// +build integ
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package gateway
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+const (
+	ASvc                 = "a" // Used by the default gateway
+	BSvc                 = "b" // Used by the custom ingress gateway
```

Comment:
I don't see this being used, maybe I missed it though

Reply:
ASvc is used when filling in the virtual service template.

I'm only using one app now, although I was contemplating making sure I could access the second if it was wired to the Istio-ingress gateway. I would have thought that was tested elsewhere, so ended up not using the second app (and could remove from the test).